dps_client/adapters/iothub_security_sas.c

```diff
@@ -469,7 +469,7 @@ BUFFER_HANDLE iothub_security_sas_sign_data(SECURITY_DEVICE_HANDLE handle, const
         BYTE data_signature[1024];
         BYTE* data_copy = (unsigned char*)data;
         /* Codes_SRS_SECURE_DEVICE_TPM_07_021: [ iothub_security_sas_sign_data shall call into the tpm to hash the supplied data value. ] */
-        uint32_t sign_len = SignData(&handle->tpm_device, &NullPwSession, data_copy, data_len, data_signature, sizeof(data_signature) );
+        uint32_t sign_len = SignData(&handle->tpm_device, &NullPwSession, data_copy, (uint32_t)data_len, data_signature, (uint32_t)sizeof(data_signature) );
```

Comment:
>(uint32_t) [](start = 85, length = 10)

Aargh.  I hate these kinds of casts.  Going from size_t to uint32_t might decrease or increase precision.  Can we update SignData to accept a size_t instead?  (I'm not looking to expand work scope -- just venting).

Reply:
this is MSR security code,  I don't know the implication of doing this since size does matter in this code.

---
In reply to: [138970748](https://github.com/Azure/azure-iot-sdk-c/pull/231#discussion_r138970748) [](ancestors = 138970748)